tests/src/test/scala/cats/tests/NonEmptyVectorTests.scala

```diff
@@ -0,0 +1,185 @@
+package cats
+package tests
+
+import cats.kernel.laws.{GroupLaws, OrderLaws}
+
+import cats.data.NonEmptyVector
+import cats.laws.discipline.{ComonadTests, FunctorTests, SemigroupKTests, FoldableTests, MonadTests, SerializableTests, CartesianTests, TraverseTests, ReducibleTests, MonadRecTests}
+import cats.laws.discipline.arbitrary._
+
+class NonEmptyVectorTests extends CatsSuite {
+  // Lots of collections here.. telling ScalaCheck to calm down a bit
+  implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    PropertyCheckConfig(maxSize = 5, minSuccessful = 20)
+
+  checkAll("NonEmptyVector[Int]", OrderLaws[NonEmptyVector[Int]].eqv)
+
+  checkAll("NonEmptyVector[Int] with Option", TraverseTests[NonEmptyVector].traverse[Int, Int, Int, Int, Option, Option])
+  checkAll("Traverse[NonEmptyVector[A]]", SerializableTests.serializable(Traverse[NonEmptyVector]))
+
+  checkAll("NonEmptyVector[Int]", ReducibleTests[NonEmptyVector].reducible[Option, Int, Int])
+  checkAll("Reducible[NonEmptyVector]", SerializableTests.serializable(Reducible[NonEmptyVector]))
+
+  implicit val iso = CartesianTests.Isomorphisms.invariant[NonEmptyVector]
+
+  // Test instances that have more general constraints
+  {
+    checkAll("NonEmptyVector[Int]", CartesianTests[NonEmptyVector].cartesian[Int, Int, Int])
+    checkAll("Cartesian[NonEmptyVector]", SerializableTests.serializable(Cartesian[NonEmptyVector]))
+  }
+
+  {
+    checkAll("NonEmptyVector[Int]", FunctorTests[NonEmptyVector].functor[Int, Int, Int])
+    checkAll("Functor[NonEmptyVector]", SerializableTests.serializable(Functor[NonEmptyVector]))
+  }
+
+  {
+    checkAll("NonEmptyVector[Int]", SemigroupKTests[NonEmptyVector].semigroupK[Int])
+    checkAll("NonEmptyVector[Int]", GroupLaws[NonEmptyVector[Int]].semigroup)
+    checkAll("SemigroupK[NonEmptyVector]", SerializableTests.serializable(SemigroupK[NonEmptyVector]))
+    checkAll("Semigroup[NonEmptyVector[Int]]", SerializableTests.serializable(Semigroup[NonEmptyVector[Int]]))
+  }
+
+  {
+    checkAll("NonEmptyVector[Int]", FoldableTests[NonEmptyVector].foldable[Int, Int])
+    checkAll("Foldable[NonEmptyVector]", SerializableTests.serializable(Foldable[NonEmptyVector]))
+  }
+
+  {
+    // Test functor and subclasses don't have implicit conflicts
+    implicitly[Functor[NonEmptyVector]]
+    implicitly[Monad[NonEmptyVector]]
+    implicitly[Comonad[NonEmptyVector]]
+  }
+
+  {
+    checkAll("NonEmptyVector[Int]", MonadTests[NonEmptyVector].monad[Int, Int, Int])
+    checkAll("Monad[NonEmptyVector]", SerializableTests.serializable(Monad[NonEmptyVector]))
+  }
+
+  {
+    checkAll("NonEmptyVector[Int]", ComonadTests[NonEmptyVector].comonad[Int, Int, Int])
+    checkAll("Comonad[NonEmptyVector]", SerializableTests.serializable(Comonad[NonEmptyVector]))
+  }
+
+  {
+    checkAll("NonEmptyVector[Int]", MonadRecTests[NonEmptyVector].monadRec[Int, Int, Int])
```

Comment:
`MonadRecTest` already runs the tests for `Monad`, `Functor`, and `Cartesian`, so you should be able to remove the test calls for the latter 3.